sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryClientBuilder.java

```diff
@@ -84,65 +114,90 @@ public ContainerRegistryClientBuilder credential(TokenCredential credential) {
     }
 
     /**
-     * Sets The HTTP httpPipeline to send requests through.
+     * Sets the HTTP pipeline to use for the service client.
+     * <p>
+     * If {@code pipeline} is set, all settings other than
+     * {@link #endpoint(String) endpoint} are ignored
+     * to build {@link ContainerRegistryAsyncClient} or {@link ContainerRegistryClient}.<br>
+     * </p>
+     *
+     * This service takes dependency on an internal policy which converts Azure token credentials into Azure Container Registry specific service credentials.
+     * In case you use your own pipeline you will have to create your own credential policy.<br>
      *
-     * @param httpPipeline the httpPipeline value.
-     * @return the ContainerRegistryBuilder.
+     * {For more information please see <a href="https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md"> Azure Container Registry Authentication </a> }.
+     *
+     * @param httpPipeline The HTTP pipeline to use for sending service requests and receiving responses.
+     * @return The updated {@link ContainerRegistryClientBuilder} object.
      */
     public ContainerRegistryClientBuilder pipeline(HttpPipeline httpPipeline) {
         this.httpPipeline = httpPipeline;
```

Comment:
Mind adding verbose level logging here when `this.httpPipeline` was previous set and is being set to null? AppConfiguration's and Search's client builders should have an example for this.